### DISCUSSION TITLE
[2022/02/15] Frontend Night

### DISCUSSION THREAD

**Discussion by jjenko** [wantedly/frontend-night#192] [Archive]
## Policy

- 資料の準備をがんばらない
- 参加・途中退席は自由
- スキップしない

話したい・聞きたいネタを書いてく ✏️

ハッシュタグ [#frontend_night](https://twitter.com/hashtag/frontend_night?src=hash)

Prev: https://github.com/wantedly/frontend-night/discussions/191 

 _Created from https://github.com/wantedly/frontend-night/discussions/179 by [issue-creator](https://github.com/rerost/issue-creator)_

**Comment by kobayang** [comment]
[Web フォントを使って contenteditable から脱出する](https://engineering.linecorp.com/ja/blog/uit-ridding-of-contenteditable/)

- textarea + overlay dom を使うことで contenteditable を回避するハックの紹介
- 絵文字フォントなど、大きさが自明ではないものはサイズだけをもつフォントを使う

**Reply by kobayang** [reply]
裏側をサイズを測るDOMにして、表側で装飾を制御するテクニックを使うことで表現力を上げることができるのは覚えておいても良さそう。社内だと、募集作成フローを実装する時に使用した。

<img src="https://user-images.githubusercontent.com/12181179/153319010-d5d38e96-e09b-4c67-add0-d81547944811.png" width="400px" />

**Reply by kobayang** [reply]
情報が二重になるので、a11y の考慮をする必要がある。
role="presentation" とかをつけると良いのかな。

**Comment by kobayang** [comment]
[2022年におけるフロントエンド開発のベースライン](https://engineering.linecorp.com/ja/blog/uit-baseline-for-front-end-development-in-2022/)

- 日本のIEのシェアはすでに2%程度。
- これからはSafariに対応する必要がある。
  - 対象とすべきSafariのVersionは過去2年分。
- WebサイトはJSを使いすぎ。特にモバイル。
  - サイズだけでなくDOM操作のコストなど複合的な観点でコストを切り詰める必要がある。

**Comment by chloe463** [comment]
[【LINE証券 FrontEnd】コンポーネントをカスタムフックで提供してみた](https://engineering.linecorp.com/ja/blog/line-securities-frontend-3/?utm_source=pocket_mylist)
[JavaScript で parseInt / parseFloat を使わない方が良い理由](http://nmi.jp/2022-02-03-dont-use-parseInt?utm_source=pocket_mylist)

**Comment by chloe463** [comment]
[Announcing TypeScript 4.6 RC](https://devblogs.microsoft.com/typescript/announcing-typescript-4-6-rc/)

**Reply by izumin5210** [reply]
breaking change になってる変更の検証
class を destructure したらメソッドは含まれない（なのでそもそもこのコードはランタイムエラーになる）
[https://www.typescriptlang.org/play?#code/MYGwhgzhAEAqAWBLAdgc2gbwFDV9CA9gLYCmACgE4EAOJFALg[…]kYQEMWBB0YwS4iS0GNEcNk8kUvVUGmRil02hkhlsUmQJGW3jQshkQA](https://www.typescriptlang.org/play?#code/MYGwhgzhAEAqAWBLAdgc2gbwFDV9CA9gLYCmACgE4EAOJFALgJ7QC80ALAEwDcWOehUgFkS9eAQAmACgCUmfnjwB6JdAB0GhdAC+WXVgBmAV2TB6iAsmgGCBADyxoJAB70SyCTAQpUAPinOAFxwctiKIKKY+MTkVLQMjAA06hoUJBD0OqzQzrxawJaEEWogBKhSAPIARgBWJGZqANYkjBBSaRkyMnmKKtAAqhAkEtD0BNAA7gQUjcmIMMgEE9BgVnRUFACEWn2UNHRM0ADkgiQiYpJH0BIE6dCLmS7zmZajjLTHFUSI9A7JAESnPbxJj-aAAH2ggJi53EEn+viOai0HXoalOsMksl4+hsBCkYQEMWBB0YwS4iS0GNEcNk8kUvVUGmRil02hkhlsUmQJGW3jQshkQA)

**Comment by chloe463** [comment]
[Core Web VitalsはWebを高速化したか？](https://postd.cc/have-core-web-vitals-made-the-web-faster/)
